### PR TITLE
Added new password rules for Excelity, HSBC, Maybank

### PR DESCRIPTION
**Pull request by CasualDeveloper**
### Changes made
Added new password rules for:
1. essportal.excelityglobal.com
Screenshot 2020-12-11 at 5 05 08 PM

2. hsbc.com.my
Screenshot 2020-12-11 at 5 08 40 PM

3. maybank2u.com.my
Screenshot 2020-12-11 at 5 22 28 PM

### Overall Checklist
- [X] I agree to the project's [Developer Certificate of Origin](https://github.com/apple/password-manager-resources/blob/main/DEVELOPER_CERTIFICATE_OF_ORIGIN.md)
- [X] The top-level JSON objects are sorted alphabetically
- [X] There are no [open pull requests](https://github.com/apple/password-manager-resources/pulls) for the same update

#### for password-rules.json
- [X] The given rule isn't particularly standard and obvious for password managers
- [X] Generated passwords have been tested from this rule using the [Password Rules Validation Tool](https://developer.apple.com/password-rules/)
- [X] Information has been included about the website's requirements (eg. screenshots, error messages, steps during experimentation, etc.)
- [X] The PR isn't documenting something that would be a common practice among password managers (e.g. minimal length of 6)

### CODE REVIEW

File: quirks/password-rules.json [hunk 1]
```diff
@@ -266,6 +269,9 @@
     "hrblock.com": {
         "password-rules": "minlength: 8; required: lower; required: upper; required: digit; required: [$#%!];"
     },
+    "hsbc.com.my": {
+        "password-rules": "minlength: 8; maxlength: 30; required: lower, digit; allowed: lower, upper, digit, [!$*-.=?@_'];"
```
Comment (Cldfire) [inline quirks/password-rules.json:273]: A couple notes about this rule:

* Any character classes that are `required` don't need to be specified in `allowed`.
* The screenshot of the site's password rule information states "It must contain at least 1 letter and 1 number". The rules `required` property will need to be `required: lower, upper; required: digit` to express that.
* The `-` character must be the first character in a custom character class.

That means the rule will end up being:

```
minlength: 8; maxlength: 30; required: lower, upper; required: digit; allowed: [-!$*.=?@_'];
```

File: quirks/password-rules.json [hunk 2]
```diff
@@ -326,6 +332,9 @@
     "marriott.com": {
         "password-rules": "minlength: 8; maxlength: 20; required: lower; required: upper; required: digit; allowed: [abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789$!#&@?%=];"
     },
+    "maybank2u.com.my": {
+        "password-rules": "minlength: 8; maxlength: 12; max-consecutive: 2; required: lower, upper, digit, special;"
```
Comment (Cldfire) [inline quirks/password-rules.json:336]: Similar to the above, the `required` property needs to be split up here (to account for the **and** relationship in the site's password rule information).

```
minlength: 8; maxlength: 12; max-consecutive: 2; required: lower; required: upper; required: digit; required: special;
```

**Review by Cldfire — CHANGES_REQUESTED** [review]

File: quirks/password-rules.json [hunk 3]
```diff
@@ -209,6 +209,9 @@
     "examservice.com.tw": {
         "password-rules": "minlength: 6; maxlength: 8;"
     },
+    "essportal.excelityglobal.com": {
```
[the remark this answers is not in the document]
Comment (CasualDeveloper) [reply quirks/password-rules.json:212]: Moved `essportal.excelityglobal.com` to above `ettoday.net`:
```suggestion
    "essportal.excelityglobal.com": {
```

File: quirks/password-rules.json (hunk 1 above)
Comment (CasualDeveloper) [reply quirks/password-rules.json:273]: Implemented your suggestion:
```suggestion
        "password-rules": "minlength: 8; maxlength: 30; required: lower, digit; allowed: lower, upper, digit, [!$*-.=?@_'];"
```

File: quirks/password-rules.json (hunk 2 above)
Comment (CasualDeveloper) [reply quirks/password-rules.json:336]: Implemented your suggestion:
```suggestion
        "password-rules": "minlength: 8; maxlength: 12; max-consecutive: 2; required: lower, upper, digit, special;"
```

**Comment by CasualDeveloper** [comment]
Suggestions implemented.

File: quirks/password-rules.json [hunk 4]
```diff
@@ -326,6 +332,9 @@
     "marriott.com": {
         "password-rules": "minlength: 8; maxlength: 20; required: lower; required: upper; required: digit; allowed: [abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789$!#&@?%=];"
     },
+    "maybank2u.com.my": {
+        "password-rules": "minlength: 8; maxlength: 12; max-consecutive: 2; required: lower; required: upper; required: digit; required: special;"
```
Comment (Cldfire) [inline quirks/password-rules.json:336]: Whoops, I missed one tidbit when I initially reviewed this!

The screenshot of the password rules for this site states:

```
Must NOT contain space
```

The `special` character class allows for spaces, though.

That means we'll have to explicitly specify the allowed special characters here (so we can omit space):

```
required: [-~!@#$%^&*_+=`|(){}[:;\"'<>,.?];
```

(I pulled that list of characters from [this page](https://developer.apple.com/documentation/security/password_autofill/customizing_password_autofill_rules).)

**Review by Cldfire — CHANGES_REQUESTED** [review]

File: quirks/password-rules.json (hunk 4 above)
Comment (CasualDeveloper) [reply quirks/password-rules.json:336]: Oops, looks like I misunderstood the following line from the [Apple Developer document](https://developer.apple.com/documentation/security/password_autofill/customizing_password_autofill_rules) then:
> You can combine these keywords to form your rule. **Duplicate property values**, empty character classes, and properties without a value are ignored.

Thanks for catching that, I'll fix it.

Comment (CasualDeveloper) [reply quirks/password-rules.json]: Ahh, I see. I didn't realise that spaces were included under the character list in the "required: special" tag.

**Comment by CasualDeveloper** [comment]
Fixed Jarek, thank you for your guidance!

**Review by Cldfire — APPROVED** [review]
Wonderful, this looks good to me now. Thank you for the contribution! 😄

**Pull request merged**